css-shared-element-transitions/index.bs

```diff
@@ -593,254 +655,307 @@ document's DOM structure.
-			viewport, if [=incoming element=] exists; otherwise,
-			|CapturedElement|&apos;s [=outgoing styles=] 'transform' property.
-
-			Let |writing-mode| and |direction| be the current value of those properties
-			on |CapturedElement|&apos;s [=incoming element=],
-			if it exists;
-			otherwise, |CapturedElement|&apos;s [=outgoing styles=] corresponding property.
-
-			At the [=user-agent origin=],
-			set |container|&apos;s 'width', 'height', 'transform', 'writing-mode', and 'direction' properties
-			to |width|, |height|, |transform|, |writing-mode|, and |direction|.
-
-		1. Create a new ''::page-transition-image-wrapper'' pseudo-element
-			with the tag |tag| and inserted as a child of |container|, if it doesn't exist.
-			Let |image wrapper| be the ''::page-transition-image-wrapper'' pseudo-element
-			with the tag |tag|.
-
-		1. If |CapturedElement| has an [=outgoing image=],
-			then create a new ''::page-transition-outgoing-image'' pseudo-element
-			with the tag |tag| and inserted as a child of |image wrapper|, if it doesn't exist.
-			Let |outgoing| be the ''::page-transition-outgoing-image'' pseudo-element with
-			the tag |tag|.
-			This pseudo-element is a [=replaced element=],
-			displaying |CapturedElement|&apos;s [=outgoing image=].
-
-			At the [=user-agent origin=],
-			set |outgoing|&apos;s 'object-view-box' property
-			to |CapturedData|&apos;s [=outgoing styles=] 'object-view-box' property.
-    
-            Issue: Which of ''xywh()''/''rect()''/''inset()'' should we use?
-
-		1. If |CapturedData| has an [=incoming element=],
-			then create be a new ''::page-transition-incoming-image''
-			pseudo-element
-			with the tag |tag|,
-			inserted as a child of |image wrapper|
-			(after |outgoing|, if it exists), if it doesn't exist.
-			Let |incoming| be the ''::page-transition-incoming-image'' pseudo-element with
-			the tag |tag|.
-			This pseudo-element is a [=replaced element=],
-			displaying the [=capture the image=]
-			of |CapturedElement|&apos;s [=incoming element=].
-
-			At the [=user-agent origin=],
-			set |incoming|&apos;s 'object-view-box' property
-			to a value that when applied to |incoming|,
-			will cause the view box to coincide with [=incoming element=]'s [=border box=]
-			in the image.
+    To <dfn>create transition pseudo-elements</dfn> for a {{SameDocumentTransition}} |transition|:
+
+    1. Let |transitionRoot| be the result of creating a new ''::page-transition'' pseudo-element.
+
+    1. [=map/For each=] |transitionTag| → |capturedElement| of |transition|'s [=SameDocumentTransition/tagged elements=]:
+
+        1. Let |container| be the result of creating a new ''::page-transition-container'' pseudo-element with the tag |transitionTag|.
+
+            Issue: "tag" should be defined/linked.
+
+        1. Append |container| to |transitionRoot|.
+
+            Issue: This should be better defined.
+                I'm not sure if pseudo-elements have defined ways to modify their DOM.
```

Comment:
I'm assuming you mean the DOM structure itself, like the ordering of the pseudo-elements. I don't think there is any defined way to do this. Generally the ordering is defined in the spec based on the pseudo-element type (like before/after).

Maybe we should add a note that because |tagged elements| is an ordered map, the DOM order for the container pseudo-elements becomes:

- Tags in the outgoing DOM based on the paint order of corresponding elements in the outgoing DOM.
- Tags in the incoming DOM (which were not in the outgoing DOM) based on the paint order of corresponding elements in the incoming DOM.

Also that devs can change the default paint order of these pseudo-elements using the z-index property.

Reply:
It's more that we have defined algorithms for adding nodes to the document, but I don't think we have that for pseudo-elements. At some point I'll speak to someone who does more CSS stuff who might know the right language to use here.